pennylane/measurements/vn_entropy.py

```diff
@@ -61,4 +64,13 @@ def circuit_entropy(x):
     .. seealso:: :func:`pennylane.qinfo.transforms.vn_entropy` and :func:`pennylane.math.vn_entropy`
     """
     wires = Wires(wires)
-    return MeasurementProcess(VnEntropy, wires=wires, log_base=log_base)
+    return _VnEntropy(VnEntropy, wires=wires, log_base=log_base)
+
+
+class _VnEntropy(StateMeasurement):
+    """Measurement process that returns the Von Neumann entropy."""
+
+    def process_state(self, state: np.ndarray, device_wires: Wires):
+        return qml.math.vn_entropy(
+            state, indices=self.wires, c_dtype=np.complex128, base=self.log_base
```

Comment:
Similar concern as with the `mutual_info` PR. I worry that we might be forcing an interface by setting the dtype here.

Reply:
Changed to "complex128". Added interface checks.